### PR TITLE
Updated documentation: added new code example and reference

### PR DESCRIPTION
**Pull request by taoism4504**
Per [Asana](https://app.asana.com/0/inbox/1200328878163177/1201051649808412/1201091064234777) task, code example was added to:
1. Authentication doc
2. Lease, Renew, and Revoke doc now contains a reference at the bottom of the page that links to the code example.

:mag: [Deploy Preview-authentication](https://vault-git-auth-go-sample-code-hashicorp.vercel.app//docs/concepts/auth#code-example)
:mag: [Deploy Preview-lease, renew, revoke](https://vault-git-auth-go-sample-code-hashicorp.vercel.app/docs/concepts/lease#lease-durations-and-renewal)

### CODE REVIEW

File: website/content/docs/concepts/lease.mdx [hunk 1]
```diff
@@ -83,3 +83,5 @@ command docs.
 This is very useful if there is an intrusion within a specific system: all
 secrets of a specific backend or a certain configured backend can be revoked
 quickly and easily.
+
+To programatically renew auth tokens, see [code example](/docs/concepts/auth#code-example).
```
Comment (yhyakuna) [inline website/content/docs/concepts/lease.mdx:87]: Wrong location.  Add this in the **Lease Durations and Renewal** section and not here. 

Also, I would word it like...

```suggestion
To implement token renewal logic in your application code, refer to the [code example in the Authentication doc](/docs/concepts/auth#code-example).
```

File: website/content/docs/concepts/lease.mdx [hunk 2]
```diff
@@ -68,6 +68,9 @@ so often.
 As a result, the return value of renewals should be carefully inspected to
 determine what the new lease is.
 
+
+To implement token renewal logic in your application code, refer to the [code example in the Authentication doc](/docs/concepts/auth#code-example).
```
Comment (yhyakuna) [inline website/content/docs/concepts/lease.mdx:72]: One last thing...  Let's put this in a call out box.

```suggestion
-> To implement token renewal logic in your application code, refer to the [code example in the Authentication doc](/docs/concepts/auth#code-example).
```

Comment (taoism4504) [reply website/content/docs/concepts/lease.mdx:72]: ok

**Review by yhyakuna — APPROVED** [review]
LGTM

**Review by digivava — APPROVED** [review]
Cool, thank you!

**Pull request merged**